### PR TITLE
Document environment variables

### LINKED ISSUE #586: Add instructions to support dotenv?
I believe dotenv is fairly popular in the Ruby community.

Here's what I did to support a "Ruby-like" dotenv with Webpack:

```
yarn add dotenv
```

```javascript
// config/webpack/dotenv.js

const dotenv = require('dotenv');

const dotenvFiles = [
  `.env.${process.env.NODE_ENV}.local`,
  '.env.local',
  `.env.${process.env.NODE_ENV}`,
  '.env'
];

function loadEnv() {
  dotenvFiles.forEach((dotenvFile) => {
    dotenv.config({ path: dotenvFile, silent: true });
  });
}

module.exports = { loadEnv };
```

```javascript
// config/webpack/shared.js

...
const dotenv = require('./dotenv');

dotenv.loadEnv();

module.exports = {

  ...

  plugins: [
    new webpack.EnvironmentPlugin(process.env),
    ...
```

And here's how to use env variables in your JS code:

```bash
# .env

STRIPE_PUBLISHABLE_KEY=...
```

```javascript
// Example JS file

Stripe.setPublishableKey(process.env.STRIPE_PUBLISHABLE_KEY);
```

Do you guys think it would be useful to add these instructions in the README? I can do a PR with the instructions above if needed.

### PR DESCRIPTION
**Pull request by jeromedalbert**
Closes #586. This PR documents environment variables and gives optional instructions to support dotenv.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1024,6 +1025,62 @@ import FooImage from 'assets/images/foo-image.png'
...
+
+You can then reference these variables in your javascript app code with
+`process.env`:
+
+```js
+console.log(process.env.FOO) // Compiles to console.log("hello")
+```
+
+You may want to store configuration in environment variables via `.env` files,
+similarly to the [dotenv Ruby gem](https://github.com/bkeepers/dotenv).
```
Comment (prathamesh-sonpatki) [inline README.md]: similar to

File: README.md [hunk 2]
```diff
@@ -1024,6 +1025,62 @@ import FooImage from 'assets/images/foo-image.png'
...
+];
+dotenvFiles.forEach((dotenvFile) => {
+  dotenv.config({ path: dotenvFile, silent: true });
+});
+
+module.exports = {
+  ...
+```
+
+**Warning:** using Foreman/Invoker and npm dotenv at the same can result in
```
Comment (prathamesh-sonpatki) [inline README.md]: at the same time

File: README.md [hunk 3]
```diff
@@ -1024,6 +1025,62 @@ import FooImage from 'assets/images/foo-image.png'
...
+confusing behavior, in that Foreman/Invoker variables can take precedence over
```
Comment (prathamesh-sonpatki) [inline README.md]: /s/can/will ? If we know that it will take precedence?

**Comment by dwightwatson** [comment]
Would it perhaps be better to move the dotenv configuration to another file (rather than `shared.js`) and then call it from there? I think it would make it clearer that it's non-standard and would allow easier removal if the developer ever change their mind.

**Comment by jeromedalbert** [comment]
Sure, in the linked issue I actually used a separate separate dotenv.js file. This file exports a loadenv function that is to be called in `shared.js`.

I could go even further and not have any function exported, so that just doing a require in `shared.js` would just work. Even easier removal if developers change their mind, although a little more magical.

**Comment by dwightwatson** [comment]
That's what I just ended up doing, and then called `require('./dotenv')` before the `model.exports`. Feels a little more plug and play.

One other thought, could the `dotenvFiles` variable just become `paths` - I think it looks a little awkward but maybe that's just me. Also gives a little more flexibility to just pass `path` into the `config` method.

```js
const dotenv = require('dotenv')

const paths = [
  `.env.${process.env.NODE_ENV}.local`,
  '.env.local',
  `.env.${process.env.NODE_ENV}`,
  '.env'
]

paths.forEach(path => {
  dotenv.config({ path, silent: true })
})

```

**Comment by gauravtiwari** [comment]
@jeromedalbert Thanks 🍰  👍

**Pull request merged**